pkg/api/scylla/validation/scylladbmanagertask_validation.go

```diff
@@ -0,0 +1,271 @@
+
+	allErrs = append(allErrs, ValidateLocalScyllaDBReference(&spec.ScyllaDBClusterRef, fldPath.Child("scyllaDBClusterRef"))...)
+
+	switch spec.Type {
+	case scyllav1alpha1.ScyllaDBManagerTaskTypeBackup:
+		if spec.Backup == nil {
+			allErrs = append(allErrs, field.Required(fldPath.Child("backup"), fmt.Sprintf("backup options are required when task type is %q", scyllav1alpha1.ScyllaDBManagerTaskTypeBackup)))
+			break
+		}
+
+		allErrs = append(allErrs, ValidateScyllaDBManagerBackupTaskOptions(spec.Backup, &options.ValidateScyllaDBManagerBackupTaskOptionsOptions, fldPath.Child("backup"))...)
+
+	case scyllav1alpha1.ScyllaDBManagerTaskTypeRepair:
+		if spec.Repair == nil {
+			allErrs = append(allErrs, field.Required(fldPath.Child("repair"), fmt.Sprintf("repair options are required when task type is %q", scyllav1alpha1.ScyllaDBManagerTaskTypeRepair)))
+			break
+		}
+
+		allErrs = append(allErrs, ValidateScyllaDBManagerRepairTaskOptions(spec.Repair, fldPath.Child("repair"))...)
+
+	default:
+		allErrs = append(allErrs, field.NotSupported(fldPath.Child("type"), spec.Type, slices.ConvertSlice(supportedScyllaDBManagerTaskTypes, slices.ToString)))
+
+	}
+
+	if spec.Type != scyllav1alpha1.ScyllaDBManagerTaskTypeBackup && spec.Backup != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("backup"), fmt.Sprintf("backup options are forbidden when task type is not %q", scyllav1alpha1.ScyllaDBManagerTaskTypeBackup)))
+	}
+
+	if spec.Type != scyllav1alpha1.ScyllaDBManagerTaskTypeRepair && spec.Repair != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("repair"), fmt.Sprintf("repair options are forbidden when task type is not %q", scyllav1alpha1.ScyllaDBManagerTaskTypeRepair)))
+	}
+
+	return allErrs
+}
+
+func ValidateScyllaDBManagerBackupTaskOptions(backupOptions *scyllav1alpha1.ScyllaDBManagerBackupTaskOptions, options *ValidateScyllaDBManagerBackupTaskOptionsOptions, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, ValidateScyllaDBManagerTaskSchedule(&backupOptions.ScyllaDBManagerTaskSchedule, fldPath)...)
+
+	if !options.IsLocationValidationDisabled {
+		if backupOptions.Location == nil || len(backupOptions.Location) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("location"), "location must not be empty"))
+		} else {
+			for i := range backupOptions.Location {
+				if len(backupOptions.Location[i]) == 0 {
+					allErrs = append(allErrs, field.Required(fldPath.Child("location").Index(i), "location must not be empty"))
+				}
+			}
+		}
+	}
+
+	return allErrs
+}
+
+func ValidateScyllaDBManagerRepairTaskOptions(repairOptions *scyllav1alpha1.ScyllaDBManagerRepairTaskOptions, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, ValidateScyllaDBManagerTaskSchedule(&repairOptions.ScyllaDBManagerTaskSchedule, fldPath)...)
+
+	if repairOptions.Intensity != nil && *repairOptions.Intensity < 0 {
```

Comment:
This could be covered on the API server side if we annotated the field with `// +kubebuilder:validation:Minimum=0`.

Reply:
we moved away from mixing the two - we use the webhook either way. Is there a benefit to doing bits of validation on the apiserver side anyway?

Reply:
The advantage is that if a user doesn't install the webhook server, they still get the validation (with fewer network hops, but it's not that important). I understand the approach where we implement everything in one place, and it makes sense if we assume that the webhook server installation is required.

Anyway, if validations are simple enough, they can be implemented with https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules and https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/ on the server side, and effectively reduce the complexity of the deployment (dropping webhook server deployment and webhook configuration). But this requires going all-in to get the benefits.

Reply:
We tried moving to CEL entirely at some point but it proved not expressive enough